packages/saltcorn-admin-models/models/backup.ts

```diff
@@ -639,17 +666,25 @@ const restore_metadata = async (dirpath: string): Promise<void> => {
 const restore = async (
   fnm: string,
   loadAndSaveNewPlugin: (plugin: Plugin) => void,
-  restore_first_user?: boolean
+  restore_first_user?: boolean,
+  password?: string
 ): Promise<string | void> => {
   const state = getState();
   state.log(2, `Starting restore to tenant ${db.getTenantSchema()}`);
 
   const tmpDir = await dir({ unsafeCleanup: true });
 
-  //unzip
-  state.log(6, `Unzipping ${fnm} to ${tmpDir}`);
-  await extract(fnm, tmpDir.path);
-  state.log(6, `Unzip done`);
+  try {
+    await extract(fnm, tmpDir.path, password);
+  } catch (error: any) {
+    if (error.requiresPassword) {
+      console.log("THROWING REQUIRES PASSWORD HERE");
+      throw { requiresPassword: true };
```

Comment:
I dont think there is any point in this try-catch. you are just rethrowing an object with the same flag set. just remove the try keyword and the entire catch clause

Reply:
Yes, totally